challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -15319,6 +15319,1381 @@
+        },
+        {
+          "text":
+            "'<code>RomanDecode(\"MDCCCXXXIX\")</code> should return <code>1839</code>.'",
+          "testString":
+            "assert.equal(RomanDecode(\"MDCCCXXXIX\"),1839,'<code>RomanDecode(\"MDCCCXXXIX\")</code> should return <code>1839</code>.');"
+        },
+        {
+          "text":
+            "'<code>RomanDecode(\"MMMMMMDCCXLV\")</code> should return <code>6745</code>.'",
+          "testString":
+            "assert.equal(RomanDecode(\"MMMMMMDCCXLV\"),6745,'<code>RomanDecode(\"MMMMMMDCCXLV\")</code> should return <code>6745</code>.');"
+        },
+        {
+          "text":
+            "'<code>RomanDecode(\"MMMMDCCCXXI\")</code> should return <code>4821</code>.'",
+          "testString":
+            "assert.equal(RomanDecode(\"MMMMDCCCXXI\"),4821,'<code>RomanDecode(\"MMMMDCCCXXI\")</code> should return <code>4821</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7fba",
+      "challengeType": 5,
+      "releasedOn": "September 6, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": ["function RomanDecode (str) {", "  // Good luck!", "}"],
+          "head": [],
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "Roots of a function",
+      "description": [
+        "Create a function that finds and returns the roots of a given function as an array.",
+        "In this task, you need to use brute force and put values in the given function to check if it is a root.",
+        "The function should take an array and 3 numbers as parameters. The array represents the function. The $ i^{th} $ element is the coefficient of $ x^i $. The numbers are lower bound, upper boand and step values. You need to start from the lower bound, increament by step value until the upper bound is reached. The values obtained should be tested if they are the roots of the given function."
+      ],
+      "solutions": [
+        "function roots (poly, lowerBound, upperBound, step) {\n  function sign(x) {\n    return (x < 0.0) ? -1 : (x > 0.0) ? 1 : 0;\n  }\n\n  function f(x) {\n    var val=0;\n    poly.forEach(function (e,i) {\n      val+=Math.pow(x,i)*e;\n    })\n    return val;\n  }\n\n  var x = lowerBound,\n    ox = x,\n    y = f(x),\n    oy = y,\n    s = sign(y),\n    os = s;\n  var roots=[]\n  for (; x <= upperBound; x += step) {\n    s = sign(y = f(x));\n    if (s == 0) {\n    } else if (s != os) {\n      var dx = x - ox;\n      var dy = y - oy;\n      var cx = x - dx * (y / dy);\n      roots.push(Math.round(cx*1000)/1000)\n    }\n    ox = x;\n    oy = y;\n    os = s;\n  }\n\n  return roots\n}\n"
+      ],
+      "tests": [
+        {
+          "text": "'<code>roots</code> should be a function.'",
+          "testString":
+            "assert(typeof roots=='function','<code>roots</code> should be a function.');"
+        },
+        
```

Comment:
`"Given a string containing uppercase characters (A-Z), compress repeated 'runs' of the same character and store the length of that run by completing the <code>encode</code> function, and complete the <code>decode</code> function to reverse the compression.",`